write-fonts/src/write.rs

```diff
@@ -44,6 +47,18 @@ pub fn dump_table<T: FontWrite + Validate>(table: &T) -> Result<Vec<u8>, Validat
     table.write_into(&mut writer);
     let mut graph = writer.finish();
     if !graph.pack_objects() {
+        //FIXME: instead of always dumping, return an error and let
+        //the caller decide
+        let time = std::time::SystemTime::now()
+            .duration_since(std::time::SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_secs();
+        let path = format!("{}_failure_{time}.dot", table.name());
+        if !std::path::Path::new(&path).exists() {
+            graph
+                .write_graph_viz(path)
+                .expect("failed to save graphviz");
+        }
```

Comment:
I think we should fix the fixme, having this unconditional happen, or even be possible to happen, feels off.

Reply:
Also I dislike that it blindly tries to write; what if I specifically want to do an entirely in-memory compile on a server? - I don't want it spewing graphviz everywhere.